core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/DataFeedServiceImpl.java

```diff
@@ -189,14 +188,27 @@ public void deleteDataFeed() throws DataFeedException {
             removePreviousDataFeed(serviceResourceResolver);
             removeCsvReport(serviceResourceResolver);
             removePendingNode(serviceResourceResolver);
-            gridResourcesCache.clearCache();
+            clearStaticDataFeed();
             serviceResourceResolver.commit();
         } catch (PersistenceException e) {
             LOG.error("Failed to delete data feed", e);
             throw new DataFeedException("Exception Deleting Data Feed");
         }
     }
 
+    private List<GridResource> getGridResourcesList() {
+        return new ArrayList<>(GRID_RESOURCE_COPY_ON_WRITE_ARRAY_LIST);
+    }
+
+    private synchronized void setGridResourcesList(List<GridResource> gridResources) {
```

Comment:
I would rather use a  synchronized block than a synchronized method. A synchronized method is aeefectuively also a synchronized lock, but it locks for access the whole service in general, which we do not need